File: 6_kyu/sum_dig_pow/sum_dig_pow.py
def sum_dig_pow(a, b):
    return [x for x in range(a, b) if sum_pow_is_x(x)]

def sum_pow_is_x(x):
    if sum([(int(y) ** (idx +1)) for idx, y in enumerate(str(x))]) == x:
        return x

File: 6_kyu/sum_dig_pow/sum_dig_pow.js
function sumDigPow(a, b) {
  let x;
  let numList = []

  for(x = a; x <= b; x++) {
    if (summedPowIsX(x)) numList.push(x)
  }

  return numList;
};

function summedPowIsX(x) {
  return String(x).split('').map((el, idx) => {
    return Math.pow(el, idx + 1)
  }).reduce((prev, curr) => prev + curr) === x
}
